tfsdk/serve_test.go

```diff
@@ -2045,50 +2040,16 @@ func TestServerPlanResourceChange(t *testing.T) {
 					}),
 				}),
 			}),
-			proposedNewState: tftypes.NewValue(testServeResourceTypeTwoType, map[string]tftypes.Value{
-				"id": tftypes.NewValue(tftypes.String, "123456"),
-				"disks": tftypes.NewValue(tftypes.List{ElementType: tftypes.Object{AttributeTypes: map[string]tftypes.Type{
-					"name":    tftypes.String,
-					"size_gb": tftypes.Number,
-					"boot":    tftypes.Bool,
-				}}}, []tftypes.Value{
-					tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{
-						"name":    tftypes.String,
-						"size_gb": tftypes.Number,
-						"boot":    tftypes.Bool,
-					}}, map[string]tftypes.Value{
-						"name":    tftypes.NewValue(tftypes.String, "my-disk"),
-						"size_gb": tftypes.NewValue(tftypes.Number, 10),
-						"boot":    tftypes.NewValue(tftypes.Bool, false),
-					}),
-				}),
-			}),
-			config:       tftypes.NewValue(testServeResourceTypeTwoType, nil),
-			resource:     "test_two",
-			resourceType: testServeResourceTypeTwoType,
-			expectedPlannedState: tftypes.NewValue(testServeResourceTypeTwoType, map[string]tftypes.Value{
-				"id": tftypes.NewValue(tftypes.String, "123456"),
-				"disks": tftypes.NewValue(tftypes.List{ElementType: tftypes.Object{AttributeTypes: map[string]tftypes.Type{
-					"name":    tftypes.String,
-					"size_gb": tftypes.Number,
-					"boot":    tftypes.Bool,
-				}}}, []tftypes.Value{
-					tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{
-						"name":    tftypes.String,
-						"size_gb": tftypes.Number,
-						"boot":    tftypes.Bool,
-					}}, map[string]tftypes.Value{
-						"name":    tftypes.NewValue(tftypes.String, "my-disk"),
-						"size_gb": tftypes.NewValue(tftypes.Number, 10),
-						"boot":    tftypes.NewValue(tftypes.Bool, false),
-					}),
-				}),
-			}),
+			proposedNewState: tftypes.NewValue(testServeResourceTypeTwoType, nil),
+			config:           tftypes.NewValue(testServeResourceTypeTwoType, nil),
+			resource:         "test_two",
+			resourceType:     testServeResourceTypeTwoType,
+			// when the config is null, that means the resource has
```

Comment:
The purpose of this test case wasn't to model null config behaviour but to ensure diags and RequiresReplace were populated properly when both are set in modifyPlanFunc. Prefer this test case to be fixed by populating the config, and we can deal with the null config case in a separate test.